extensions/ql-vscode/test/vscode-tests/cli-integration/run-cli.test.ts

```diff
@@ -23,10 +24,14 @@ describe("Use cli", () => {
   let cli: CodeQLCliServer;
   let supportedLanguages: string[];
 
+  let logSpy: jest.SpiedFunction<BaseLogger["log"]>;
+
   beforeEach(async () => {
     const extension = await getActivatedExtension();
     cli = extension.cliServer;
     supportedLanguages = await cli.getSupportedLanguages();
+
+    logSpy = jest.spyOn(cli.logger, "log");
```

Comment:
I had to make the CodeQLCLIServer's `logger` property `public readonly` for this to work. I think that's okay? But shout if not!

(PS: I initially tried `logSpy = jest.spyOn(extLogger, "log")` but that didn't work.)

Reply:
Would something like this work? I know we're trying to avoid `as any` in the code, but I'm not sure what is better...

```suggestion
    logSpy = jest.spyOn((cli as any).logger, "log");
```